backtrace-library/src/main/java/backtraceio/library/base/BacktraceBase.java

```diff
@@ -445,6 +479,72 @@ public boolean addBreadcrumb(String message, Map<String, Object> attributes, Bac
         return database.getBreadcrumbs().addBreadcrumb(message, attributes, type, level);
     }
 
+    /**
+     * Enable backtrace metrics support
+     * @param settings  Backtrace metrics settings
+     */
+    public void enableMetrics(BacktraceMetricsSettings settings) {
+        // For tracking crash-free sessions we need to add
+        // application.session and application.version to Backtrace attributes
+        String sessionId = UUID.randomUUID().toString();
+        this.attributes.put("application.session", sessionId);
+        try {
+            this.attributes.put("application.version", this.context.getPackageManager()
+                    .getPackageInfo(this.context.getPackageName(), 0).versionName);
+        } catch (PackageManager.NameNotFoundException e) {
+            BacktraceLogger.e(LOG_TAG, "Could not resolve package version information for Backtrace metrics");
+        }
+
+        metrics = new BacktraceMetrics(context, attributes, settings);
+        metrics.startMetricsEventHandlers(backtraceApi);
+        metrics.sendStartupEvent();
+    }
+
+    public boolean addUniqueEvent(String attributeName) {
```

Comment:
why do we need to expose this interface once again? Why user can't use `backtraceClient.Metrics.addUniqueEvent` instead? it's not a better interface for our users - with `metrics`  prefix we scope methods that belong to something else - we did similar thing with breadcrumb.

Reply:
Ah okay that sounds pretty cool. We didn't do this for breadcrumbs in `backtrace-android` but we probably should have. I will try to change it to this approach